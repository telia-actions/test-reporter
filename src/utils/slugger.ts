// Returns HTML element id and href link usable as manual anchor links
// This is needed because Github in check run summary doesn't automatically
// create links out of headings as it normally does for other markdown content
import {ReportOptions} from '../report/get-report'

export function slug(name: string, options: ReportOptions): {id: string; link: string} {
  const slugId = name
    .trim()
    .replace(/_/g, '')
    .replace(/[./\\]/g, '-')
    .replace(/[^\w-]/g, '')

  const id = `user-content-${slugId}`
  // When using the Action Summary for display, links must include the "user-content-" prefix.
  const link = options.useActionsSummary ? `#${id}` : `#${slugId}`
  return {id, link}
}
