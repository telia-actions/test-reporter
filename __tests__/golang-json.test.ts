import * as fs from 'fs'
import * as path from 'path'

import {GolangJsonParser} from '../src/parsers/golang-json/golang-json-parser'
import {ParseOptions} from '../src/test-parser'
import {getReport} from '../src/report/get-report'
import {normalizeFilePath} from '../src/utils/path-utils'

describe('golang-json tests', () => {
  it('report from ./reports/dotnet test results matches snapshot', async () => {
    const fixturePath = path.join(__dirname, 'fixtures', 'golang-json.json')
    const outputPath = path.join(__dirname, '__outputs__', 'golang-json.md')
    const filePath = normalizeFilePath(path.relative(__dirname, fixturePath))
    const fileContent = fs.readFileSync(fixturePath, {encoding: 'utf8'})

    const opts: ParseOptions = {
      parseErrors: true,
      trackedFiles: ['calculator.go', 'calculator_test.go']
    }

    const parser = new GolangJsonParser(opts)
    const result = await parser.parse(filePath, fileContent)
    expect(result).toMatchSnapshot()

    const report = getReport([result])
    fs.mkdirSync(path.dirname(outputPath), {recursive: true})
    fs.writeFileSync(outputPath, report)
  })
})
