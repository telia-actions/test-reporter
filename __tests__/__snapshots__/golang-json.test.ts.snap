// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`golang-json tests report from ./reports/dotnet test results matches snapshot 1`] = `
TestRunResult {
  "path": "fixtures/golang-json.json",
  "suites": [
    TestSuiteResult {
      "groups": [
        TestGroupResult {
          "name": null,
          "tests": [
            TestCaseResult {
              "error": undefined,
              "name": "TestPassing",
              "result": "success",
              "time": 60,
            },
            TestCaseResult {
              "error": {
                "details": "calculator_test.go:19: expected 1+1 = 3, got 2
",
                "message": "calculator_test.go:19: expected 1+1 = 3, got 2
",
              },
              "name": "TestFailing",
              "result": "failed",
              "time": 890,
            },
            TestCaseResult {
              "error": {
                "details": "calculator_test.go:76: caught panic: runtime error: integer divide by zero
",
                "message": "calculator_test.go:76: caught panic: runtime error: integer divide by zero
",
              },
              "name": "TestPanicInsideFunction",
              "result": "failed",
              "time": 0,
            },
            TestCaseResult {
              "error": {
                "details": "calculator_test.go:76: caught panic: bad stuff
",
                "message": "calculator_test.go:76: caught panic: bad stuff
",
              },
              "name": "TestPanicInsideTest",
              "result": "failed",
              "time": 0,
            },
            TestCaseResult {
              "error": {
                "details": "calculator_test.go:45: skipping test
",
                "message": "calculator_test.go:45: skipping test
",
              },
              "name": "TestSkipped",
              "result": "skipped",
              "time": 940,
            },
            TestCaseResult {
              "error": {
                "details": "",
                "message": "",
              },
              "name": "TestCases",
              "result": "failed",
              "time": 2250,
            },
          ],
        },
        TestGroupResult {
          "name": "TestCases",
          "tests": [
            TestCaseResult {
              "error": undefined,
              "name": "1_+_2_=_3",
              "result": "success",
              "time": 400,
            },
            TestCaseResult {
              "error": undefined,
              "name": "4_+_7_=_11",
              "result": "success",
              "time": 460,
            },
            TestCaseResult {
              "error": {
                "details": "calculator_test.go:67: expected 2 + 3 = 4, got 5
",
                "message": "calculator_test.go:67: expected 2 + 3 = 4, got 5
",
              },
              "name": "2_+_3_=_4",
              "result": "failed",
              "time": 90,
            },
            TestCaseResult {
              "error": {
                "details": "calculator_test.go:67: expected 1 / 2 = 1, got 0
",
                "message": "calculator_test.go:67: expected 1 / 2 = 1, got 0
",
              },
              "name": "1_/_2_=_1",
              "result": "failed",
              "time": 920,
            },
            TestCaseResult {
              "error": undefined,
              "name": "9_/_3_=_3",
              "result": "success",
              "time": 340,
            },
            TestCaseResult {
              "error": undefined,
              "name": "14_/_7_=_2",
              "result": "success",
              "time": 40,
            },
          ],
        },
      ],
      "name": "_/home/james_t/git/test-reporter/reports/go",
      "totalTime": undefined,
    },
  ],
  "totalTime": undefined,
}
`;
